### PR TITLE
Generate static test functions & wrappers

### PR DESCRIPTION
**Pull request by misch7**
Fix `missing-prototype` errors in `mbedtls/tests/suites`

Required for: https://github.com/Mbed-TLS/mbedtls/pull/9189

### CODE REVIEW

**Review by gilles-peskine-arm — DISMISSED** [review]
LGTM

File: scripts/generate_test_code.py
Comment (misch7) [reply scripts/generate_test_code.py:655]: > This could be better to modify in line 647 above.
> 
> ```
>         args, local_vars, args_dispatch = parse_function_arguments(
>             line)
>         code += line.replace('void', 'static void', 1)
>         break
> ```

This feels kind of mingled; I wanted to be consistent with these lines below:

https://github.com/Mbed-TLS/mbedtls-framework/blob/3d25f35d68528317ab58db19485727b0ed9a53cc/scripts/generate_test_code.py#L657-L659

https://github.com/Mbed-TLS/mbedtls-framework/blob/3d25f35d68528317ab58db19485727b0ed9a53cc/scripts/generate_test_code.py#L664

IMHO more readable and easier to maintain, without interfering with the matching logic above.

> 
> But assuming we want to only replace a `void ` in the begining of the file we could achieve the same with:
> 
> ```
> code += re.sub(r'^void .', 'static void ',line)
> ```

This breaks my local build again because not all functions are made static. Did you test it?

> 
> Either way works

Shall we keep it like before?

Comment (minosgalanakis) [reply scripts/generate_test_code.py:655]: Yes it feels trivial change to block the pr. I will look how to optimise the generate_test_code in the future.

**Comment by minosgalanakis** [comment]
@misch7 we recently merged a major/significant pr  #37 on the framework. Could you please rebase this one on top of it? 

Main Repository PR [9189](https://github.com/Mbed-TLS/mbedtls/pull/9189) will also need to have the farmework [pointer ](https://github.com/misch7/mbedtls/commit/3236dd9a0744912f5ddc3fc62a1a07184ca7c842)updated to target the post rebase target of this PR.

**Comment by misch7** [comment]
> @misch7 we recently merged a major/significant pr #37 on the framework. Could you please rebase this one on top of it?
> 
> Main Repository PR [9189](https://github.com/Mbed-TLS/mbedtls/pull/9189) will also need to have the farmework [pointer ](https://github.com/misch7/mbedtls/commit/3236dd9a0744912f5ddc3fc62a1a07184ca7c842)updated to target the post rebase target of this PR.

@minosgalanakis Done and I've also resolved the new merge conflicts in the main repo.

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM after rebase

**Pull request merged**
